frontend/src/BoxplotWithControls.tsx

```diff
@@ -97,62 +104,86 @@ function useOptions(facet: keyof Row) {
   return [ui.useIntern(options), nodes] as const
 }
 
-function useVisibleSidebar(facet: string, facet_values: string[]) {
-  const all_selected = Object.fromEntries(facet_values.map(v => [v, true]))
+function useVisibleSidebar(facet: string, facet_values_unsorted: string[]) {
+  const facet_values = React.useMemo(
+    () =>
+      facet.match(/cell/)
+        ? utils.sort_cells(facet_values_unsorted)
+        : utils.sort_tumors(facet_values_unsorted),
+    [facet, facet_values_unsorted]
+  )
+
+  const [store] = useStore({visible: facet_values})
 
-  const [visible_facets, facet_boxes] = ui.useCheckboxes(facet_values, all_selected, utils.pretty)
+  // React.useLayoutEffect(() => {
+  // store.update({visible: facet_values})
+  // }, [utils.str(facet_values)])
 
   const [show, set_show] = React.useState(true)
 
+  const facet_boxes = (
+    <CheckboxRow store={store} values={facet_values} column="visible" labelBy={utils.pretty} />
+  )
+
+  const value = Object.fromEntries(store.get().visible.map(v => [v, true]))
+
+  const [delayed_value, enqueue_delayed_value] = ui.useDelayed(400, value)
+
+  const loading = !utils.equal(value, delayed_value)
+
+  React.useEffect(() => {
+    enqueue_delayed_value(value)
```

Comment:
why does the value need to be delayed? It's probably correct - but an explanation would help!